odoo/addons/base/models/ir_ui_view.py

```diff
@@ -2708,7 +2708,7 @@ def _get_records_action(self, **kwargs):
         if len(self) == 0:
             length_dependent = {'views': [(False, 'form')]}
         elif len(self) == 1:
-            length_dependent = {'views': [(False, 'form')], 'res_id': self.id}
+            length_dependent = {'views': [(False, 'form')], 'res_id': self._origin.id}
```

Comment:
There is another suggestion for that part in this PR:
https://github.com/odoo/odoo/pull/166733/files#diff-dfebe5a93e1b8880e88268b024be4c6f106d144b20298d7bb6c4ae09a18bafd0R2830

I think we should decide right now what is the best approach to handle new records in this method.

Reply:
`_origin` was initially (quickly) suggested by @william-andre . 

[This](https://github.com/odoo/odoo/pull/166733/files#r1634704519) instead is backed by @xmo-odoo and @yosuanicolaus .

Reply:
I'm fine with the https://github.com/odoo/odoo/pull/166733 version

Reply:
Both are good.
The other PR is changing more stuff in this function so it is okay to change for pattern matching, but I wouldn't do it here since it is more diff for nothing more

Reply:
Let's just put a comment for clarity in case of `NewId`s.
Pattern matching enforces the idea that there are different strategies needed by different input. I don't think the diff size is a big issue, this is a `base` function used for every model in Odoo. 

I'd say `base` is the realm of `xmo` more than ours, but any r+er can just take a decision 😄 

```suggestion
                'domain': [('id', 'in', ids)]

        # `ids` will silently take care of eventual `NewId`s
        match self.ids:
            case []:
                length_dependent = {'views': [(False, 'form')]}
            case [id]:
                length_dependent = {'views': [(False, 'form')], 'res_id': id}
            case ids:
                length_dependent = {
                     'views': [(False, 'list'), (False, 'form')],
                     'domain': [('id', 'in', ids)]
```

Reply:
Yosua's version has been merged. Time to move on.